packages/enzyme-test-suite/test/shared/hooks/custom.jsx

```diff
@@ -0,0 +1,147 @@
+          <Counter {...props}>
+            {(val) => {
+              Object.assign(returnVal, val);
+              return null;
+            }}
+          </Counter>,
+        );
+        return returnVal;
+      }
+
+      it('useCounter', () => {
+        const counterData = setup();
+        counterData.increment();
+        expect(counterData).to.have.property('count', 1);
+        counterData.decrement();
+        expect(counterData).to.have.property('count', 0);
+      });
+
+      it('useCounter with initialCount', () => {
+        const counterData = setup({ initialCount: 2 });
+        counterData.increment();
+        expect(counterData).to.have.property('count', 3);
+        counterData.decrement();
+        expect(counterData).to.have.property('count', 2);
+      });
+
+      it('useCounter with step', () => {
+        const counterData = setup({ step: 2 });
+        counterData.increment();
+        expect(counterData).to.have.property('count', 2);
+        counterData.decrement();
+        expect(counterData).to.have.property('count', 0);
+      });
+
+      it('useCounter with step and initialCount', () => {
+        const counterData = setup({ step: 2, initialCount: 5 });
+        counterData.increment();
+        expect(counterData).to.have.property('count', 7);
+        counterData.decrement();
+        expect(counterData).to.have.property('count', 5);
+      });
+    });
+
+    // todo: enable when useEffect works in the shallow renderer
+    describeIf(!isShallow, 'custom hook: formInput simulate', () => {
+      function useFormInput(initialValue = '') {
+        const [value, setValue] = useState(initialValue);
+
+        return {
+          value,
+          onChange(e) {
+            setValue(e.target.value);
+          },
+        };
+      }
+
+      function Input(props) {
+        return (
+          <div>
+            <input {...props} />
+          </div>
+        );
+      }
+
+      function ControlledInputWithEnhancedInput({ searchSomething }) {
+        const search = useFormInput();
+
+        useEffect(
+          () => {
+            searchSomething(search.value);
+          },
+          [search.value],
+        );
+
+        return <Input {...search} />;
+      }
+
+      function ControlledInputWithNativeInput({ searchSomething }) {
+        const search = useFormInput();
+
+        useEffect(
+          () => {
+            searchSomething(search.value);
+          },
+          [search.value],
+        );
+
+        return <input {...search} />;
+      }
+
+      it('does not succeed without act', () => {
+        const spy = sinon.spy();
+        const wrapper = Wrap(<ControlledInputWithNativeInput searchSomething={spy} />);
+        // Works with Act
+        // act(() => {
+        wrapper.simulate('change', { target: { value: 'foo' } });
+        // });
+
+        expect(spy.withArgs('foo')).to.have.property('callCount', 1);
+      });
+
+      // TODO: Need to evaluate as per issue raised
```

Comment:
@chenesan what's needed here?

Reply:
This is fixed after we wrap `.simulate` with act in #2135 . I'll fix the tests.